pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -187,15 +190,27 @@ private KinesisMessageBatch buildKinesisMessageBatch(KinesisPartitionGroupOffset
     return new KinesisMessageBatch(messages, offsetOfNextBatch, endOfShard);
   }
 
-  private String getShardIterator(String shardId, String sequenceNumber) {
+  private String getShardIterator(String shardId, String sequenceNumber,
+      CommonConstants.Segment.Realtime.StreamContinuationMode continuationMode) {
     GetShardIteratorRequest.Builder requestBuilder =
         GetShardIteratorRequest.builder().streamName(_config.getStreamTopicName()).shardId(shardId);
-    if (sequenceNumber != null) {
-      requestBuilder = requestBuilder.startingSequenceNumber(sequenceNumber)
-          .shardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER);
-    } else {
-      requestBuilder = requestBuilder.shardIteratorType(_config.getShardIteratorType());
+
+    switch (continuationMode) {
+      case RESUME: {
+        if (sequenceNumber != null) {
+          requestBuilder = requestBuilder.startingSequenceNumber(sequenceNumber).shardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER);
+        } else {
+          requestBuilder = requestBuilder.shardIteratorType(_config.getShardIteratorType());
+        }
+        break;
+      }
+      case INITIALIZE: {
+        requestBuilder = requestBuilder.shardIteratorType(_config.getShardIteratorType());
+        break;
+      }
+      default: //
```

Comment:
This is bad practice to leave unexpected mode unhandled. Throw exception instead